vpr/src/route/connection_router.cpp

```diff
@@ -1113,17 +1100,25 @@ static inline bool relevant_node_to_target(const RRGraphView* rr_graph,
     return false;
 }
 
-static inline void update_router_stats(const DeviceContext& device_ctx,
-                                       const RRGraphView* rr_graph,
+#ifdef VTR_ENABLE_DEBUG_LOGGING
+static inline void update_router_stats(const RRGraphView* rr_graph,
```

Comment:
Consider changing the parameter order to mandatory ones first, then ones that are only needed with debug logging.